src/protocol/coap.rs

```diff
@@ -0,0 +1,130 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum Method {
+    Get,
+    Put,
+    Post,
+    Delete,
+    Patch,
+    #[serde(rename = "iPATCH")]
+    Ipatch,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, Default)]
+pub struct BlockWiseTransferParameters {
+    #[serde(rename = "cov:block2SZX")]
+    block_2szx: Option<u16>,
+    #[serde(rename = "cov:block1SZX")]
+    block_1szx: Option<u16>,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, Default)]
+// #[serde(rename_all = "camelCase")]
+pub struct Form {
+    #[serde(rename = "cov:method")]
+    method: Option<Method>,
+    #[serde(rename = "cov:blockwise")]
+    blockwise: Option<BlockWiseTransferParameters>,
+    #[serde(rename = "cov:qblockwise")]
+    qblockwise: Option<BlockWiseTransferParameters>,
+    #[serde(rename = "cov:hopLimit")]
+    hop_limit: Option<u8>,
+}
+
+use crate::extend::ExtendableThing;
```

Comment:
Uhm, why a use statement in the middle of other things? It is a bit unexpected -- if you remove the line and re-import it using Rust Analizer, it will be placed in a "good" position.